### ISSUE TITLE
x/airdrop/keeper: Keeper.decompress is too complex

### ISSUE THREAD

**Issue by elias-orijtech** [ingenuity-build/quicksilver#220]
https://github.com/ingenuity-build/quicksilver/blob/4ea3aa77d56b674c7189835064fda16f6bebf8da/x/airdrop/keeper/proposal_handler.go#L74-L100

can be replaced with (untested):

```
func (k Keeper) decompress(data []byte) ([]byte, error) {
	zr, err := zlib.NewReader(bytes.NewReader(data))
	if err != nil {
		return nil, err
	}
	defer zr.Close()
	return io.ReadAll(zr)
}
```

CC @odeke-em

**Comment by zanicar** [comment]
No, this introduces a DoS vulnerability via a decompression bomb...

**Comment by elias-orijtech** [comment]
It's true that the replacement is vulnerable to decompression bombs. However, so was the original because there is no limit to the amount of decompressed data. In fact, `io.ReadAll` contains code similar to `Keeper.decompress`: https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/io/io.go;l=654.

**Comment by odeke-em** [comment]
Yeah, @zanicar as @elias-orijtech just pointed out, I too don't see how the original code prevents a zip bomb except by using bytes.Buffer that grows incrementally versus slice reallocations in 1 shot. The first link you provided ensures that if for example optimizations like sendfile are applied that io.Copy can take advantage of them, but in the instance of the code in this repo, it doesn't.

**Comment by joe-bowman** [comment]
This has been closed in earlier audit remediation fixes.
